### DISCUSSION TITLE
Do we have an updated version of swagger page code for Next13?

### DISCUSSION THREAD

**Discussion by JohnVersus** [jellydn/next-swagger-doc#560] [Q&A] [answered]
I tried updating the code with the below code but it only threw more errors saying `Module not found: Can't resolve 'fs'`. Similar errors getting thrown even after installing `fs`. Is it possible to update the swagger docs UI code to work with the next js 13? 
```
import { createSwaggerSpec } from 'next-swagger-doc';
import dynamic from 'next/dynamic';
import 'swagger-ui-react/swagger-ui.css';

const SwaggerUI = dynamic<{
  spec: any;
}>(import('swagger-ui-react'), { ssr: false });

const getDocs = async () => {
  const spec: Record<string, any> = createSwaggerSpec({
    definition: {
      openapi: '3.0.0',
      info: {
        title: 'Next Swagger API Example',
        version: '1.0',
      },
    },
  });
  return spec;
};

async function ApiDoc() {
  const spec = await getDocs();
  return <SwaggerUI spec={spec} />;
}

export default ApiDoc;

```

**Comment by jellydn** [comment]
Yeah, it should be fine @JohnVersus I've just tested with example app. Could you take a look at example folder? That's using nextjs v13. Thanks.

**Comment by JohnVersus** [comment]
You tried with the code that I shared and it worked?

Are you referring to this folder?
https://github.com/jellydn/next-swagger-doc/tree/main/example

**Comment by JohnVersus** [comment]
Ahh!! Just saw the example folder. Seems it is using the pages folder. I had trouble using it with app folder.  

I was able to make it work with the pages folder without any changes, but not with the app folder.

**Comment by jellydn** [comment]
I see. I will give it a try soon :) Thanks.

**Comment by jellydn** [comment, marked as answer]
FYI @JohnVersus Added nextjs 13 example with `app` folder https://github.com/jellydn/next-swagger-doc/tree/main/examples/next13-example

**Comment by JohnVersus** [comment]
Awesome!! Thanks🙌. 

Tested it just now. It is working great. 🏆
